homeassistant/components/screenlogic/__init__.py

```diff
@@ -34,29 +35,35 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-REQUEST_REFRESH_DELAY = 1
+REQUEST_REFRESH_DELAY = 2
+HEATER_COOLDOWN_DELAY = 6
+
+# These seem to be constant across all controller models
+PRIMARY_CIRCUIT_IDS = [500, 505]  # [Spa, Pool]
 
 PLATFORMS = ["switch", "sensor", "binary_sensor", "climate", "light"]
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Screenlogic component."""
-    domain_data = hass.data[DOMAIN] = {}
-    domain_data[DISCOVERED_GATEWAYS] = await async_discover_gateways_by_unique_id(hass)
+    hass.data[DOMAIN] = {}
```

Comment:
You could drop `async_setup` completely and do 

`hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator`

below

Reply:
Actually you probably need to do `hass.data.setdefault(DOMAIN, {})` at the top of `async_setup_entry`

Reply:
Will be in second commit. Need to do a little more so `async_get_connect_info` is not expecting the domain data to be there yet. Saving the discovered gateways is not needed anymore anyways.

Reply:
https://github.com/home-assistant/core/pull/60466/commits/0cf4dc99b8427a447ea0365e207e06edc48ea138